test_scripts/build_with_buildcache.sh

```diff
@@ -60,6 +60,8 @@ ln -s "${BUILDCACHEEXE}" "${SYMLINKSDIR}/clang++${EXESUFFIX}"
 ln -s "${BUILDCACHEEXE}" "${SYMLINKSDIR}/cl${EXESUFFIX}"
 export PATH="${SYMLINKSDIR}:${BUILDCACHEDIR}:${PATH}"
 
+export CMAKE_OSX_ARCHITECTURES="x86_64;arm64"
```

Comment:
I'd prefer to have this as an environment setting in `.github/workflows/build_and_test.yml` instead, e.g:

```diff
@@ -113,6 +113,7 @@ jobs:
       env:
         CC: "${{ matrix.config.cc }}"
         CXX: "${{ matrix.config.cxx }}"
+        CMAKE_OSX_ARCHITECTURES: "x86_64;arm64"
       run: |
         ../test_scripts/run_file_lock_stresstest.sh
         ../test_scripts/build_with_buildcache.sh
```